### ISSUE TITLE
Mouse event won't be processed by Window EVER in any area once popup menu covered

### ISSUE THREAD

**Issue by forrestjgq** [isl-org/Open3D#4843]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

In Open3D, once a menu is activated, that area will never be processed by Window's child widget any more. The reason why this happen is because the popup Window is in ImGui buffer and won't be destroyed after popup menu closed, and in Window.cpp there is a judgement to avoid delivering mouse event to child widget while there is a NOT Hidden popup Window contains this mouse event point:

```
    // Some ImGUI widgets have popup windows, in particular, the color
    // picker, which creates a popup window when you click on the color
    // patch. Since these aren't gui::Widgets, we don't know about them,
    // and will deliver mouse events to something below them. So find any
    // that would use the mouse, and if it isn't a toplevel child, then
    // eat the event for it.
    if (e.type == MouseEvent::BUTTON_DOWN || e.type == MouseEvent::BUTTON_UP) {
        ImGuiContext* context = ImGui::GetCurrentContext();
        for (auto* w : context->Windows) {
            if (!w->Hidden && w->Flags & ImGuiWindowFlags_Popup) {    <== MenuItem_XX always exists and always not Hidden
                Rect r(int(w->Pos.x), int(w->Pos.y), int(w->Size.x),
                       int(w->Size.y));
                if (r.Contains(e.x, e.y)) {
                    bool weKnowThis = false;
                    for (auto child : impl_->children_) {
                        if (child->GetFrame() == r) {
                            weKnowThis = true;
                            break;
                        }
                    }
                    if (!weKnowThis) {
                        // This is not a rect that is one of our children,
                        // must be an ImGUI internal popup. Eat event.
                        PostRedraw();
                        return;
                    }
                }
            }
        }
    }
```
The menu popup window MenuItem_XX always exists and is NOT Hidden.

### Steps to reproduce the bug

```python
Run Open3D with any point cloud, activate any menu and then close.
Try click and drag mouse starting from the area that popup menu ever covered, you'll find it will never be processed(Point cloud does not rotate).
```

### Error message

No message

### Expected behavior

Closed popup should be ignored

### Open3D, Python and System information

```markdown
- Operating system: Ubuntu 20.04
- Python version: Python 3.9
- Open3D version: 0.15.2
- System architecture: x86 
- Is this a remote workstation?: no
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 9.3
```

**Comment by forrestjgq** [comment]
I think I solved this issue by #4844

**Comment by errissa** [comment]
Thanks for reporting this. What an interesting bug. I had never noticed this but it was easy to reproduce once you pointed it out. I've approved #4844 and it should be merged soon.
